### DISCUSSION TITLE
My CI/CD breaks as Tag v4.1.4 doesn't point to PR merge commit for master branch. It points to commit in develop.

### DISCUSSION THREAD

**Discussion by marsteel** [netbox-community/netbox#17773] [General]
### Deployment Type

Self-hosted

### Triage priority

N/A

### NetBox Version

v4.1.3

### Python Version

3.12

### Steps to Reproduce

I have a GitHub workflow in my repo to track NetBox master branch everyday. It checks the tags and find out the latest release by parsing the latest tag.  Then I do code review again and merged to my local main branch.

It works well for previous releases. It stopped working with v4.1.4. 

```
          UPSTREAM_TAG=$(git ls-remote --tags | grep -h $(git rev-parse --short ${{ env.upstream_remote_name }}/${{ env.upstream_branch_name }} ) | awk 'END{print}' | awk '{print $2}' | cut -d'/' -f 3)
          UPSTREAM_HEAD_COMMIT=$(git rev-parse --short ${{ env.upstream_remote_name }}/${{ env.upstream_branch_name }} )
```

The tag for latest version v4.1.4 doesn't point the commit of "Merge pull request" in master branch. Or the latest commit in master branch is not in the repo tag list. There is inconsistence.

With `git ls-remote --tags` or `git ls-remote --tags https://github.com/netbox-community/netbox.git` the tags can be feteched.
The tag v4.1.4 points to d2cbdfe7d742f0d2db7989ed27cde466c8366dea in develop branch 
While other tags pointed to "Merge pull request" in master branch.

Tag v4.1.4 should point to 6ea0c0c in master branch

```
7bc0d34196323ac992d7ec80b1caa48e6094d88d        refs/tags/v4.1.0
Merge pull request #17350 from netbox-community/develop                 <--- commit in master branch
0e34fba92223348e0bf4375b8d380324ff5e1beb        refs/tags/v4.1.1
Merge pull request #17478 from netbox-community/develop                 <--- commit in master branch
ead6e637f4aecc4717b10c71f9140c94040da264        refs/tags/v4.1.2
Merge pull request #17626 from netbox-community/develop                 <--- commit in master branch
6ea0c0c3e910d1104fd0fbe5e6cd07198862d1fa        refs/tags/v4.1.3
Merge pull request #17658 from netbox-community/develop                 <--- commit in master branch
d2cbdfe7d742f0d2db7989ed27cde466c8366dea        refs/tags/v4.1.4
Release v4.1.4                                            <--- commit in develop branch
```

### Expected Behavior

Tag v4.1.4 should point to 6ea0c0c in master branch

### Observed Behavior

Tag v4.1.4 points to d2cbdfe7d742f0d2db7989ed27cde466c8366dea in develop branch

**Comment by candlerb** [comment]
> Tag v4.1.4 should point to https://github.com/netbox-community/netbox/commit/6ea0c0c3e910d1104fd0fbe5e6cd07198862d1fa in master branch

Surely not, since that commit is tag v4.1.3.  I think you mean 4deb6e59

> Tag v4.1.4 points to https://github.com/netbox-community/netbox/commit/d2cbdfe7d742f0d2db7989ed27cde466c8366dea in develop branch

```
% git log origin/develop
commit dbc52dc6c7e726b2f25451f03547b8e7accd07ea (origin/develop, origin/HEAD)
Author: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Wed Oct 16 05:02:10 2024 +0000

    Update source translation strings

commit 4deb6e5968a245bff7fa2a2ffb13b941b49a0f9d (HEAD -> master, origin/master)
Merge: 6ea0c0c3e d2cbdfe7d
Author: Jeremy Stretch <jstretch@netboxlabs.com>
Date:   Tue Oct 15 13:59:27 2024 -0400

    Merge pull request #17763 from netbox-community/develop

    Release v4.1.4

commit d2cbdfe7d742f0d2db7989ed27cde466c8366dea (tag: v4.1.4)
Author: Jeremy Stretch <jstretch@netboxlabs.com>
Date:   Tue Oct 15 13:42:25 2024 -0400

    Release v4.1.4

```

The develop branch is currently one commit ahead of master, which is one commit ahead of v4.1.4

You're probably right that for consistency v4.1.4 would point to 4deb6e59. However in practice it doesn't make any difference since the content is the same:

```
% git diff 4deb6e59 d2cbdfe7
%
```

It's rather hard to change it now, since slipping the tag in the main repository doesn't slip the tag for anyone who has already picked it up. See: https://git-scm.com/docs/git-tag#_on_re_tagging

**Reply by candlerb** [reply]
Just to add: is the problem you're trying to solve that you want to know the *tag name* of the latest release? Given that the master branch is the latest release (by definition), I'd suggest:

```
% git tag --merged origin/master --sort=-committerdate | head -1
v4.1.4
```

(that is: the most recent tag on the specified branch)

**Reply by marsteel** [reply]
@candlerb I have replace my Get Tag command with your version.  Your command works. Issue resolved. Thanks.
```
#Get Tag
#UPSTREAM_TAG=$(git ls-remote --tags | grep -h $(git rev-parse --short ${{ env.upstream_remote_name }}/${{ env.upstream_branch_name }} ) | awk 'END{print}' | awk '{print $2}' | cut -d'/' -f 3)
#https://github.com/netbox-community/netbox/discussions/17773#discussioncomment-10963368
UPSTREAM_TAG=$(git tag --merged ${{ env.upstream_remote_name }}/${{ env.upstream_branch_name }} --sort=-committerdate | head -1)
```
